src/shared/modules.mjs

```diff
@@ -854,6 +855,137 @@ const generateEventListenResponse = json => {
+    let methodResultSchema = {}
+    if (methodResult.schema.anyOf === anyOfTypes) {
+        let anyOfType = generateAnyOfSchema(anyOf, title, summary)
+        let index = 0
+        if (isEventMethod(method)) {
+            index = (method.result.schema.anyOf || method.result.schema.oneOf).indexOf(getPayloadFromEvent(method))
+        }
+        else {
+            index = (method.result.schema.anyOf || method.result.schema.oneOf).indexOf(anyOfType)
+        }
+        if (method.result.schema.anyOf) {
+            methodResultSchema["anyOf"] = Object.assign([], method.result.schema.anyOf)
+            methodResultSchema.anyOf[index] = anyOfType.schema
+        }
+        else if (method.result.schema.oneOf) {
+            methodResultSchema["oneOf"] = Object.assign([], method.result.schema.oneOf)
+            methodResultSchema.oneOf[index] = anyOfType.schema
+        }
+        else {
+            methodResultSchema = anyOfType.schema
+        }
+    }
+    return methodResultSchema
+}
+
+const createPolymorphicMethods = (method, json) => {
+    let anyOfTypes
+    let methodParams = []
+    let methodResult = Object.assign({}, method.result)
+    method.params.forEach(p => {
+        if (p.schema) {
+            let param = getAnyOfSchema(p, json)
+            if (param.schema.anyOf && anyOfTypes) {
+                //anyOf is allowed with only one param in the params list
+                throw `WARNING anyOf is repeated with param:${p}`
+            }
+            else if (param.schema.anyOf) {
+                anyOfTypes = param.schema.anyOf
+            }
+            methodParams.push(param)
+        }
+    })
+    let foundAnyOfParams = anyOfTypes ? true : false
+
+    if (isEventMethod(method)) {
+        methodResult.schema = getPayloadFromEvent(method)
+    }
+    methodResult = getAnyOfSchema(methodResult, json)
+    let foundAnyOfResult = methodResult.schema.anyOf ? true : false
+    if (foundAnyOfParams === true && foundAnyOfResult === true) {
+        throw `WARNING anyOf is already with param schema, it is repeated with ${method.name} result too`
+    }
+    else if (foundAnyOfResult === true) {
+        anyOfTypes = methodResult.schema.anyOf
+    }
+    let polymorphicMethodSchemas = []
+    //anyOfTypes will be allowed either in any one of the params or in result
+    if (anyOfTypes) {
+        let polymorphicMethodSchema = {
+            name: {},
+            tags: {},
+            summary: `${method.summary}`,
+            params: {},
+            result: {},
+            examples: {}
+        }
+        anyOfTypes.forEach(anyOf => {
+
+            let localized = localizeDependencies(anyOf, json)
+            let title = localized.title || localized.name || ''
+            let summary = localized.summary || localized.description || ''
+            polymorphicMethodSchema.title = method.name
+            polymorphicMethodSchema.name = `${method.name}${title}`
```

Comment:
Could we change this to:

```javascript
polymorphicMethodSchema.name = foundAnyOfParams ? `${method.name}With${title}` : `${method.name}${title}`
```
That way, users of our API will know that the method is being forked for parameters if the word "With" is in the name, and forked for results otherwise.

Reply:
done